### PR TITLE
Sp 1065

### LINKED ISSUE #1065: Provide e2e tests to validate the different formats of generic adapters
### Body

A generic adapter can have different formats (e.g. json, csv, ...).
An e2e test should validate that events represented in those formats can be parsed correctly into the internal event representation.

### StreamPipes Committer

I acknowledge that I am a maintainer/committer of the Apache StreamPipes project.

### PR DESCRIPTION
**Pull request by tenthe**
### Purpose

Fixes #1065 by adding e2e tests for the different formats that can be used in the adapters.
Also enabled the live preview feature for all formats.

### Remarks

This PR also removes the underscore dependency and replaces it with jackson-dataformat-xml, which was already registered in the pom.
I also had to adjust some of the parser implementations. This PR provides the basis for further refactoring of those parsers.
PR introduces (a) breaking change(s): no

PR introduces (a) deprecation(s): no

### CODE REVIEW

**Review by bossenti — APPROVED** [review]

File: streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/format/geojson/GeoJsonParser.java [hunk 1]
```diff
@@ -85,66 +85,85 @@ public void parse(InputStream data, EmitBinaryEvent emitBinaryEvent) throws Pars
...
       if (point.getCoordinates().hasAltitude()) {
         eventProperties.add(getEventPropertyGeoJson("altitude", point.getCoordinates().getAltitude(), SO.ALTITUDE));
+        sampleValues.put("altitude", new GuessTypeInfo("java.lang.Double", point.getCoordinates().getAltitude()));
       }
 
     } else if (geoFeature.getGeometry() instanceof LineString) {
       LineString lineString = (LineString) geoFeature.getGeometry();
       eventProperties.add(JsonEventProperty.getEventProperty("coorindatesLineString", lineString.getCoordinates()));
-
+      sampleValues.put("coorindatesLineString", new GuessTypeInfo("java.lang.Double", lineString.getCoordinates()));
```
Comment (dominikriemer) [inline streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/format/geojson/GeoJsonParser.java:133]: is this a type and should be named `coordinates`?

File: streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/format/geojson/GeoJsonParser.java [hunk 2]
```diff
@@ -85,66 +85,85 @@ public void parse(InputStream data, EmitBinaryEvent emitBinaryEvent) throws Pars
...
     } else if (geoFeature.getGeometry() instanceof Polygon) {
       Polygon polygon = (Polygon) geoFeature.getGeometry();
       eventProperties.add(JsonEventProperty.getEventProperty("coorindatesPolygon", polygon.getCoordinates()));
-
+      sampleValues.put("coorindatesPolygon", new GuessTypeInfo("java.lang.Double", polygon.getCoordinates()));
     } else if (geoFeature.getGeometry() instanceof MultiPoint) {
       MultiPoint multiPoint = (MultiPoint) geoFeature.getGeometry();
       eventProperties.add(JsonEventProperty.getEventProperty("coorindatesMultiPoint", multiPoint.getCoordinates()));
-
+      sampleValues.put("coorindatesMultiPoint", new GuessTypeInfo("java.lang.Double", multiPoint.getCoordinates()));
```
Comment (dominikriemer) [inline streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/format/geojson/GeoJsonParser.java:141]: same for all types

File: streampipes-extensions/streampipes-connect-adapters-iiot/pom.xml [hunk 3]
```diff
@@ -127,6 +127,12 @@
             <groupId>org.apache.plc4x</groupId>
             <artifactId>plc4j-driver-s7</artifactId>
             <scope>runtime</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>com.fasterxml.woodstox</groupId>
+                    <artifactId>woodstox-core</artifactId>
+                </exclusion>
```
Comment (dominikriemer) [inline streampipes-extensions/streampipes-connect-adapters-iiot/pom.xml:134]: Does this exclusion have any impact on the S7 driver?

**Review by dominikriemer — CHANGES_REQUESTED** [review]

File: streampipes-extensions/streampipes-connect-adapters-iiot/pom.xml (hunk 3 above)
Comment (tenthe) [reply streampipes-extensions/streampipes-connect-adapters-iiot/pom.xml:134]: Hopefully not, we have to validate it with a PLC as soon as it is merged. This is included due to some convergence errors

File: streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/format/geojson/GeoJsonParser.java (hunk 1 above)
Comment (tenthe) [reply streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/format/geojson/GeoJsonParser.java:133]: Thanks, you are right. It seems to be a typo.

**Comment by tenthe** [comment]
@dominikriemer I have addressed your comments. I also extracted some common strings into a new class `GeoJsonConstants`.
We should improve the implementation further, but this should reduce the risk of typos for now.

**Review by dominikriemer — APPROVED** [review]
Thanks!

**Pull request merged**